packages/rs-drive-abci/src/abci/server.rs

```diff
@@ -0,0 +1,81 @@
+//! This module implements ABCI application server.
+//!
+use crate::error::execution::ExecutionError;
+use crate::{config::PlatformConfig, error::Error, platform::Platform, rpc::core::CoreRPCLike};
+use drive::grovedb::Transaction;
+use std::fmt::Debug;
+use std::sync::RwLock;
+
+/// AbciApp is an implementation of ABCI Application, as defined by Tenderdash.
+///
+/// AbciApp implements logic that should be triggered when Tenderdash performs various operations, like
+/// creating new proposal or finalizing new block.
+pub struct AbciApplication<'a, C> {
+    /// Platform
+    pub platform: &'a Platform<C>,
+    /// The current transaction
+    pub transaction: RwLock<Option<Transaction<'a>>>,
+}
+
+/// Start ABCI server and process incoming connections.
+///
+/// Should never return.
+pub fn start<C: CoreRPCLike>(config: &PlatformConfig, core_rpc: C) -> Result<(), Error> {
+    let bind_address = config.abci.bind_address.clone();
+
+    let platform: Platform<C> =
+        Platform::open_with_client(&config.db_path, Some(config.clone()), core_rpc)?;
+
+    let abci = AbciApplication::new(&platform)?;
+
+    let server =
+        tenderdash_abci::start_server(&bind_address, abci).map_err(super::AbciError::from)?;
+
+    loop {
+        tracing::info!("waiting for new connection");
+        match server.handle_connection() {
```

Comment:
What I don't really understand is whether there is any condition when we just crash the server. It seems to me that we're gonna be running even if there's no DB connection anymore